File: Scripts/Obtain_daily_mean_distance.py
from Functions import *

parameters = {"path output": "../Output/",
              "path data": "../Data/",
              "path information": "../Information/",
              "file distances": "Distance_between.csv",
              "file output": "Daily_mean_distance.csv",
              "useless columns": ["Usuario_Id",
                                  "Genero",
                                  "Viaje_Id",
                                  "Año_de_nacimiento",
                                  "Inicio_del_viaje",
                                  "Fin_del_viaje"]
              }

files = obtain_filenames(parameters["path data"])
period = obtain_period_from_filenames(files)
dates = obtain_consecutive_dates_from_period(period)
daily_data_output = create_daily_dataframe(dates,
                                           header="Mean distance",
                                           use_float=True)
distance_data = read_data(parameters["path output"],
                          parameters["file distances"],
                          use_index=True)
for file in files:
    print("Analizando archivo {}".format(file))
    data = read_data(parameters["path data"],
                     file)
    distances = distance_algorithm(data,
                                   parameters["useless columns"],
                                   distance_data,
                                   distance_data.index)
    data = distances.data
    daily_data = data.resample("D").mean()
    del data
    for date in daily_data.index:
        index = date.date()
        value = daily_data["Distance"][date]
        daily_data_output.loc[index, "Mean distance"] = value
    del daily_data
daily_data_output.to_csv("{}{}".format(parameters["path output"],
                                       parameters["file output"]))

File: Scripts/Obtain_daily_mean_distance.R
suppressPackageStartupMessages({
    library(dplyr)
})
source("Functions.R")

path_data <- "../Data/with_distances/"
path_output <- "../Output/"
distance_file <- paste(
    path_output,
    "Distance_between.csv",
    sep = ""
)
files <- list.files(path_data)
dates <- obtain_dates_consecutive(files)
daily_data <- data.frame(
    distance = matrix(0.01,
        nrow = length(dates)
    ),
    row.names = dates
)
distance_data <- read.csv(distance_file)
header <- colnames(distance_data)
id_list <- distance_data$X0
files <- c("2014_12.csv")
for (file in files) {
    print(paste("Analizando archivo", file))
    filename <- paste(path_data,
        file,
        sep = ""
    )
    data <- read.csv(filename)
    data <- clean_data(data, id_list)
    data <- subset(data, distance != 0)
    data$Inicio_del_viaje <- as.Date(data$Inicio_del_viaje)
    mean <- data %>%
        group_by(Inicio_del_viaje) %>%
        dplyr::summarize(Mean = mean(distance))
    mean$Inicio_del_viaje <- as.Date(mean$Inicio_del_viaje)
    for (index in rownames(mean)) {
        index <- as.integer(index)
        date <- mean$Inicio_del_viaje[index]
        distance <- mean$Mean[index]
        index <- match(date, dates)
        daily_data$distance[index] <- distance
    }
    rm(data)
}
daily_data <- na.omit(daily_data)
write.table(daily_data,
    "output.csv",
    sep = ",",
    quote = FALSE,
    col.names = "Date,Mean distance",
)